File: app/src/main/java/mishka/mingeo/view/pumpinglist/PumpingListViewModel.java
package mishka.mingeo.view.pumpinglist;


import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import java.util.List;

import mishka.mingeo.data.datamanager.DataManager;
import mishka.mingeo.data.model.Pumping;

public class PumpingListViewModel extends ViewModel {
    private DataManager dataManager;
    private LiveData<List<Pumping>> pumpings;

    public PumpingListViewModel(DataManager dataManager) {
        this.dataManager = dataManager;
        pumpings = dataManager.getPumpings();
    }

    public LiveData<List<Pumping>> getPumpings(){
        return pumpings;
    }

    public void onAddPumpingClick() {
        dataManager.createPumping();
    }
}

File: app/src/main/java/mishka/mingeo/view/pumpinglist/PumpingListViewModel.kt
package mishka.mingeo.view.pumpinglist


import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider

import mishka.mingeo.data.datamanager.DataManager
import mishka.mingeo.data.model.Pumping
import javax.inject.Inject

class PumpingListViewModel(private val dataManager: DataManager) : ViewModel() {
    val pumpings: LiveData<List<Pumping>> = dataManager.pumpings

    fun onCreatePumpingClick() = dataManager.createPumping()


    class Factory @Inject constructor(val dataManager: DataManager) : ViewModelProvider.NewInstanceFactory() {
        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
            return PumpingListViewModel(dataManager) as T
        }
    }
}
